src/ActiveBackground/Confetti.ts

```diff
@@ -25,20 +22,20 @@ const COLORS = [
   ['#ffd200', '#b06c00'],
 ]
 
-class Confetti implements IActiveBackgroundPattern {
-  canvas: HTMLCanvasElement
-  context: CanvasRenderingContext2D | null
+export class Confetti implements ActiveBackgroundPattern {
+  private readonly canvas: HTMLCanvasElement
```

Comment:
Mark everything as `private` and `readonly` by default. This will help `tsc` determine what is used / unused. Only remove modifiers (and make it explicitly `public`) when necessary.